[master] feat: allow configurable destination for netbox ext_pillar data

raddessi opened the pull request:
### What does this PR do?
This adds support to allow the user to configure in what pillar key the netbox ext_pillar data gets populated in. For example I already use the `netbox` pillar for server and client configuration and there are a few overlapping keys there. Personally I would prefer to leave the existing config as-is and then move the minion data in to the pillar location `netbox:minion`.


### What issues does this PR fix or reference?
I didn't open an issue for this since it seemed like a trivial addition for me to just add and ask for opinions on afterwards.

### Previous Behavior
The netbox minion data would always be populated in the `netbox` pillar.

### New Behavior
The netbox minion data will be populated in the `netbox` pillar by default but the user can now specify the pillar location they would like it placed in to, optionally.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

raddessi commented on tests/pytests/unit/pillar/test_netbox.py:
```diff
@@ -2318,6 +2318,60 @@ def test_when_we_retrieve_everything_successfully_then_return_dict(
+    default_kwargs["site_details"] = True
+    default_kwargs["site_prefixes"] = True
+    default_kwargs["proxy_return"] = True
+    default_kwargs["connected_devices"] = True
+    default_kwargs["destination_pillar_key"] = "netbox:minion"
+
```
I just copied the full test so this could be cleaned up for sure. I can add a minimal response fixture that is dedicated for this test if wanted.

raddessi replied:
Scratch that, I have now copied the fixture data also. In theory we could deduplicate this a little now if you would like.

raddessi commented on salt/pillar/netbox.py:
```diff
@@ -88,6 +88,14 @@
+destination_pillar_key: ``netbox``
+    .. versionadded:: 3007.0
+
+    A colon-delimited compound key string of nested pillar keys in which the
+    data should be populated. This lets put the data in the pillar at
```
What's a cleaner way to say this?

twangboy replied:
How about this from ChatGPT:
A colon-delimited string representing a nested path within the pillar where the data should be placed. This allows flexibility in organizing pillar data. For example, you can store the data under netbox:minion or an-arbitrary-location, which is useful if you already use the netbox pillar and need to avoid key conflicts within it.

raddessi replied:
I'll update this PR tonight, thanks!

raddessi commented:
@ggiesen you may want to comment here? Thanks!

raddessi commented:
the logic doesn't work, I will update

raddessi commented:
Looks like the current test failure may be unrelated?

dwoz committed:
feat: allow configurable destination for netbox ext_pillar data

Add destination_pillar_key option to the netbox ext_pillar to let users
configure which pillar key the netbox data is populated in. Defaults to
'netbox' for backward compatibility. Supports colon-delimited nested keys
such as 'netbox:minion'.

Addresses review feedback: improved docstring per twangboy suggestion.
